### ISSUE TITLE
pathlib.Path.read_text should include a newline argument

### ISSUE THREAD

**Issue by treyhunner** [python/cpython#110745]
# Feature or enhancement

### Proposal:

Support for a `newline` argument was added to the `write_text` method in Python 3.10 (issue #67894).

I've been using this method but I've found a need for a `newline` method for `read_text` as well.

Here's the scenario where this would be handy (namely reading and writing the original newlines):

```python
from pathlib import Path

path = Path("my_file.txt")

contents = path.read_text(newline="")

contents = perform_some_operation_on(contents)

path.write_text(contents, newline="")
```

### Has this already been discussed elsewhere?

This is a minor feature, which does not need previous discussion elsewhere

### Links to previous discussion of this feature:

I have not found any prior discussions or mentions of `newline` argument being supported for `read_text` outside of the initial pathlib API discussion (https://github.com/python/cpython/issues/64417#issuecomment-1093641934).

### Linked PRs
* gh-110880

**Comment by Okabe-Junya** [comment]
Thanks for all the great software!

I would like to work on this issue. This is my first contribution to cpython, but I will do my best!

**Comment by barneygale** [comment]
Resolved in #110880. Thank you very much @Okabe-Junya!
